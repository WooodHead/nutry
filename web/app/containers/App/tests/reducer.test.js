// import { fromJS, List } from 'immutable';

import appReducer from '../reducer';

import {
  login,
  loginSuccess,
  loginFailure,
  searchComplete } from '../actions';

describe('Container [App] - reducers', () => {
  let state;
  beforeEach(() => {
  });
  it('should return the initial state', () => {
    expect(appReducer(undefined, {})).toMatchSnapshot();
  });
  it('should handle the loginSuccess action correctly', () => {
    expect(appReducer(state, login())).toMatchSnapshot();
  });
  it('should handle the loginSuccess action correctly', () => {
    const token = 'd95f8aed95f8aed95f8aed95f8aed95f8ae';

    expect(appReducer(state, loginSuccess(token))).toMatchSnapshot();
  });
  it('should handle the loginFailure action correctly', () => {
    const err = 'error message';
    expect(appReducer(state, loginFailure(err))).toMatchSnapshot();
  });
  it('should handle the loginFailure action correctly', () => {
    const searchResults = [];
    expect(appReducer(state, searchComplete(searchResults))).toMatchSnapshot();
  });
});
