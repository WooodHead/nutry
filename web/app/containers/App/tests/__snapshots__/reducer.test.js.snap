// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [App] - reducers should handle the loginFailure action correctly 1`] = `
Immutable.Map {
  "loggedIn": false,
  "searchResults": Immutable.List [],
  "loading": true,
  "error": true,
  "username": false,
  "token": false,
}
`;

exports[`Container [App] - reducers should handle the loginFailure action correctly 2`] = `
Immutable.Map {
  "loggedIn": false,
  "searchResults": Array [],
  "loading": false,
  "error": false,
  "username": false,
  "token": false,
}
`;

exports[`Container [App] - reducers should handle the loginSuccess action correctly 1`] = `
Immutable.Map {
  "loggedIn": false,
  "searchResults": Immutable.List [],
  "loading": true,
  "error": false,
  "username": undefined,
  "token": false,
}
`;

exports[`Container [App] - reducers should handle the loginSuccess action correctly 2`] = `
Immutable.Map {
  "loggedIn": true,
  "searchResults": Immutable.List [],
  "loading": true,
  "error": false,
  "username": false,
  "token": "d95f8aed95f8aed95f8aed95f8aed95f8ae",
}
`;

exports[`Container [App] - reducers should return the initial state 1`] = `
Immutable.Map {
  "loggedIn": false,
  "searchResults": Immutable.List [],
  "loading": true,
  "error": false,
  "username": false,
  "token": false,
}
`;
